src/lib/vocab/entity/entityServices.ts

```diff
@@ -40,13 +45,21 @@ export const ReadEntitiesPaginatedService: ServiceByName['ReadEntitiesPaginated'
 	action: ReadEntitiesPaginated,
 	transaction: false,
 	perform: async ({repos, params}) => {
-		const {actor, source_id, pageSize, pageKey} = params;
+		const {actor, source_id, pageSize, pageKey, sibling} = params;
+		log.debug('checking pagiated entities for ', source_id);
 		const {hub_id} = await repos.space.findByEntity(source_id);
 		await checkHubAccess(actor, hub_id, repos);
 
-		const ties = await repos.tie.filterBySourceIdPaginated(source_id, pageSize, pageKey);
+		const pageTies = await repos.tie.filterBySourceIdPaginated(source_id, pageSize, pageKey);
+		let siblingTies: Tie[] = [];
+		if (sibling !== 'none' && pageTies.length > 0) {
+			const pageEntityIds = pageTies.map((t) => t.dest_id);
+			siblingTies = await repos.tie.filterSiblingsByEntityId(pageEntityIds, sibling);
+		}
+		const ties = pageTies.concat(siblingTies);
```

Comment:
was confusing to see it instantiate an array just to concat it, here's a possible improvement:

```suggestion
		let siblingTies: Tie[] | null = null;
		if (sibling !== 'none' && pageTies.length > 0) {
			const pageEntityIds = pageTies.map((t) => t.dest_id);
			siblingTies = await repos.tie.filterSiblingsByEntityId(pageEntityIds, sibling);
		}
		const ties = siblingTies ? pageTies.concat(siblingTies) : pageTies;
```

if we wanted to be 100% safe but less efficient we could end that line with `pageTies.slice();`